aten/src/ATen/native/cudnn/MHA.cpp

```diff
@@ -0,0 +1,327 @@
+                                                          .set_attn_scale(attn_scale)
+                                                          .set_dropout(dropout_probability, seed, offset);
+    // Optional bias in flash attention is only supported 8.9.3 onwards
+    if (cudnnGetVersion() >= 8904) {
+        //scaled_dot_product_flash_attention_options.set_alibi_mask(true);
+    }
+
+    auto seq_q  = mha_graph->tensor(fe::graph::Tensor_attributes()
+                                    .set_name("seq_q")
+                                    .set_dim({b, 1, 1, 1})
+                                    .set_stride({1, 1, 1, 1})
+                                    .set_data_type(fe::DataType_t::INT32));
+    auto seq_kv = mha_graph->tensor(fe::graph::Tensor_attributes()
+                                    .set_name("seq_kv")
+                                    .set_dim({b, 1, 1, 1})
+                                    .set_stride({1, 1, 1, 1})
+                                    .set_data_type(fe::DataType_t::INT32));
+
+    //if (cudnnGetVersion() >= 8903) {
+    //    scaled_dot_product_flash_attention_options.set_bias(bias)
+    //        .set_padding_mask(true)
+    //        .set_seq_len_q(seq_q)
+    //        .set_seq_len_kv(seq_kv);
+    //}
+
+
+    auto [O, Stats] = mha_graph->sdpa(Q, K, V, scaled_dot_product_flash_attention_options);
+    O->set_output(true).set_dim(std::vector<int64_t>(o.sizes().data(), o.sizes().data() + o.sizes().size())).set_stride(std::vector<int64_t>(o.strides().data(), o.strides().data() + o.strides().size()));
+
+    if (Stats) {
+        Stats->set_output(true).set_data_type(fe::DataType_t::FLOAT);
+    }
+
+    AT_CUDNN_FRONTEND_CHECK(mha_graph->validate());
+    AT_CUDNN_FRONTEND_CHECK(mha_graph->build_operation_graph(handle));
+    AT_CUDNN_FRONTEND_CHECK(mha_graph->create_execution_plans({fe::HeurMode_t::A}));
+    AT_CUDNN_FRONTEND_CHECK(mha_graph->check_support(handle));
+    AT_CUDNN_FRONTEND_CHECK(mha_graph->build_plans(handle));
+
+    return std::make_tuple(mha_graph, Q, K, V, attn_scale, seed, offset, O, Stats);
+}
+
+void
+run_cudnn_LLM_fprop(int64_t b,
+                    int64_t h,
+                    int64_t s_q,
+                    int64_t s_kv,
+                    int64_t d,
+                    float scaling_factor,
+                    bool return_softmaxstats,
+                            bool is_causal,
+                    double dropout_probability,
+                    const Tensor& q,
+                    const Tensor& k,
+                    const Tensor& v,
+                    Tensor& softmaxstats,
+                    Tensor& o,
+                    Tensor& dropoutseed,
+                    Tensor& dropoutoffset) {
+
+    cudnnHandle_t handle = getCudnnHandle();
+    o = at::empty_strided({b, h, s_q, d}, {s_q * h * d, d, h * d, 1}, q.options());
+    if (return_softmaxstats) {
+      // TODO(eqy): fix strides
```

Comment:
I assume this stride fix is only needed for backward support right? so is okay for this PR?